engine/docker/docker.go

```diff
@@ -525,3 +537,33 @@ func (e *Docker) setProxyInDockerDaemon(ctx context.Context, pipelineConfig *spe
 		}
 	}
 }
+
+// SoftStop stops the container giving them a 30 seconds grace period. The signal sent by ContainerStop is SIGTERM.
+// After the grace period, the container is killed with SIGKILL.
+// After all the containers are stopped, they are removed only when the status is not "running" or "removing".
+func (e *Docker) SoftStop(ctx context.Context, name string) {
+	timeout := 30 * time.Second
```

Comment:
Can we move this value to the top of the file in the constants?

Reply:
We can't pass the const address to the method so I can't put it in the const block